### DISCUSSION TITLE
Is tss-react is future proof for react 18

### DISCUSSION THREAD

**Discussion by amitdvs** [garronej/tss-react#58] [General]
current working on the web app, made with react 17 and material UI 4 and we are trying to upgrade it to MUI 5.
we are heavily using makeStyles that is depreciated to MUI 5

 is tss-react is future proof for react18?

**Comment by garronej** [comment]
I haven't investigated yet but the general idea is that TSS is built on top [emotion](https://emotion.sh/docs/introduction) which is much bigger than I and MUI itself is built on top of it.  
I see @Andarist (he is emotion) is already on it [here](https://github.com/reactwg/react-18/discussions/110).  
I am pretty confident that, comes the time, emotion will publish specific instructions for making SSR work with React 18.  
I will then update [`tss-react/nextJs.tsx`](https://github.com/garronej/tss-react/blob/main/src/nextJs.tsx), `tss-react/gatsby` (upcoming) and the [specific documentation](https://docs.tss-react.dev/ssr).  

What I can say for sure is that I am committed to keeping TSS relevant but I can't guarantee anything beyond "I will do my best and save some time to work on it."

**Comment by garronej** [comment]
It's now compatible with React 18

**Reply by amitdvs** [reply]
Thanks for your dedication to makeStyles, tss-react save lots of my time and work from the MUI upgrade journey from v4 to v5.
 
**long live makeStyles!**

**Reply by garronej** [reply]
Thank you for the kind words, truly appreciated.

**Reply by garronej** [reply]
Hi @amitdvs,  
I just discovered [this github program](https://stars.github.com/) and since you where so nice to me I though maybe you'd be willing to suggest my name, to help me in my opensourcer career.  
Please don't feel obligated.  
Good day!

**Reply by amitdvs** [reply]
Done

**Reply by garronej** [reply]
Thank you very very much!
